### PR TITLE
Add the size of failed allocation in the panic message

### PR DESCRIPTION
**Pull request by dureuill**
# Pull Request

## Related issue
Not fixing, but might help to debug https://github.com/meilisearch/meilisearch/issues/3018

## What does this PR do?
- add the size information to the allocation that failed in `EntryBoundAlignedBuffer`, so that we know which value was passed to the allocator on failure.

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: src/sorter.rs [hunk 1]
```diff
@@ -347,7 +347,11 @@ impl EntryBoundAlignedBuffer {
         let size = (size + entry_bound_size - 1) / entry_bound_size * entry_bound_size;
         let layout = Layout::from_size_align(size, align_of::<EntryBound>()).unwrap();
         let ptr = unsafe { alloc(layout) };
-        assert!(!ptr.is_null(), "the allocator is unable to allocate that much memory");
+        assert!(
+            !ptr.is_null(),
+            "the allocator is unable to allocate that much memory ({:#x} bytes requested)",
```
Comment (Kerollmops) [inline src/sorter.rs:352]:
```suggestion
            "the allocator is unable to allocate that much memory ({} bytes requested)",
```

I would prefer that we format that in decimal, easier to read for me :)

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Don't forget to change the test accordingly :)

File: src/sorter.rs (hunk 1 above)
Comment (dureuill) [reply src/sorter.rs]: Done exactly as suggested

**Review by Kerollmops — APPROVED** [review]
Looks perfect to me. Thank you @dureuill ❤️

**Pull request merged**